xdsl/dialects/builtin.py

```diff
@@ -101,9 +101,10 @@
     from xdsl.printer import Printer
 
 
-DYNAMIC_INDEX = -1
+DYNAMIC_INDEX = -9_223_372_036_854_775_808
```

Comment:
Seems a little neater:
```suggestion
DYNAMIC_INDEX = -(2 **63)
```